lib/render.js

```diff
@@ -186,52 +190,42 @@ MapnikSource.prototype._renderMetatile = function(options, callback) {
     // Add reference to the source allowing debug/stat reporting to be compiled.
     options.source = source;
 
-    process.nextTick(function() {
-        // acquire can throw if pool is draining
```

Comment:
As you upgraded `generic-pool`, we need to review if acquire can throw errors. If not, please remove this ugly `try/catch`

Reply:
It won't throw errors now, [just reject the promise](https://github.com/coopernurse/node-pool/blob/72b31a434c7b05ad879b2ace9830a9aa9fbba002/lib/Pool.js#L434).

But `map.render` can throw depending on the input. It's one of those functions that for input related errors (like invalid types in options) throws, but for other stuff (errors during the render), returns the error in the callback. So we still need the try-catch.

Reply:
IMHO, this one of biggest pains of `node-mapnik`, it's an anti-pattern in Node.js ecosystem. When we have the chance, I'd like to change that behavior: any asynchronous function MUST propagate `errors` via `callback` OR rejecting the `promise`.